WalletWasabi/WabiSabi/Client/CoinJoinTrackingData.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.WabiSabi.Client;
+
+public record CoinJoinTrackingData : IDisposable
```

Comment:
So the name contains "data" suffix but then the `record` contains `CoinJoinClient` and that actually does the coinjoining (sends HTTP requests, etc.)

So isn't this something more like:

```suggestion
public record CoinJoinProcess : IDisposable
```
?

-or-

```suggestion
public record CoinJoinTracker : IDisposable
```
?

I'm also not sure why `record` is used here as it looks to me more like a `class` to me. Records were imo introduced to encapsulate data (e.g. `Person`) and not processes.

```suggestion
public class CoinJoinTracker : IDisposable
```

Reply:
I mean something that basically means "single CJ process" (hopefully that's what it is).

Reply:
I agree.